lib/kafka_ex.ex

```diff
@@ -46,6 +46,53 @@ defmodule KafkaEx do
   end
 
   @doc """
+  Returns a KafkaEx worker child specification.
+
+  This child specification can be used to host KafkaEx workers outside of the
+  KafkaEx application itself, for example in your own application supervision
+  tree. Use together with the application option `disable_default_worker` set
+  to `true` to make sure no processes are started by the KafkaEx
+  application.
+
+  It takes the same arguments as create_worker/2.
+
+  Returns {:error, error_description} on invalid arguments
+
+  ## Example
+
+  ```elixir
+  iex> KafkaEx.worker_spec(:pr) # where :pr is the name of the worker created
+  {:ok,
+   {KafkaEx.Server,
+    {KafkaEx.Server, :start_link,
+     [[uris: [{"localhost", 9092}, {"localhost", 9093}, {"localhost", 9094}],
+       consumer_group: "kafka_ex"], :pr]}, :permanent, 5000, :worker,
+    [KafkaEx.Server]}}
+
+  iex> KafkaEx.worker_spec(:pr, [uris: [{"localhost", 9092}], consumer_group: "foo", sync_timeout: 2000])
+  {:ok,
+   {KafkaEx.Server,
+    {KafkaEx.Server, :start_link,
+     [[uris: [{"localhost", 9092}], consumer_group: "foo", sync_timeout: 2000],
+      :pr]}, :permanent, 5000, :worker, [KafkaEx.Server]}}
+
+  iex> KafkaEx.worker_spec(:pr, consumer_group: nil)
+  {:error, :invalid_consumer_group}
+
+  ```
+
+  """
+  @spec worker_spec(atom, KafkaEx.worker_init) :: {:ok, Supervisor.Spec.spec()} | {:eror, any}
+  def worker_spec(name, worker_init \\ []) do
```

Comment:
I'm not a big fan of verbs if the function returns a simple "item" (i.e. `get_value` or `return_result`). I think in this case, it is not a huge operation, therefore I'd like to avoid the heavy verb `build`. The other alternative should be `worker_child_spec` and here I'm of the opinion that it is a bit redundant. If you're after a child spec, you know you will put it in a supervisor, and a worker is normally not a supervisor, which implies it is just a normal child. The name "worker" in KafkaEx is unfortunate because it is the same as `Supervisor.Spec.worker`, so actually the proper canonical name should be `worker_worker_spec`. :smile: